src/detectors/HCAL/RawCalorimeterHit_factory_LFHCALRawHits.h

```diff
@@ -0,0 +1,99 @@
+// Copyright 2022, David Lawrence
+// Subject to the terms in the LICENSE file found in the top-level directory.
+//
+
+#ifndef _RawCalorimeterHit_factory_LFHCALRawHits_h_
+#define _RawCalorimeterHit_factory_LFHCALRawHits_h_
+
+#include <random>
+
+#include <JANA/JEvent.h>
+#include <JANA/JFactoryT.h>
+#include <services/geometry/dd4hep/JDD4hep_service.h>
+#include <algorithms/calorimetry/CalorimeterHitDigi.h>
+#include <edm4hep/SimCalorimeterHit.h>
+#include <edm4hep/RawCalorimeterHit.h>
+#include <Evaluator/DD4hepUnits.h>
+#include <services/log/Log_service.h>
+#include <extensions/spdlog/SpdlogExtensions.h>
+
+
+
+class RawCalorimeterHit_factory_LFHCALRawHits : public JFactoryT<edm4hep::RawCalorimeterHit>, CalorimeterHitDigi {
+
+public:
+
+    //------------------------------------------
+    // Constructor
+    RawCalorimeterHit_factory_LFHCALRawHits() {
+        SetTag("LFHCALRawHits");
+        m_log = japp->GetService<Log_service>()->logger(GetTag());
+    }
+
+    //------------------------------------------
+    // Init
+    void Init() override {
+        auto app = GetApplication();
+
+        // Set default values for all config. parameters in CalorimeterHitDigi algorithm
+        m_input_tag = "LFHCALHits";
+        u_eRes = {};
+        m_tRes = 0.0; // in ns
+        m_capTime = 100 ; // given in ns
```

Comment:
If this is compared to .getTime(), this needs explicit units:
```suggestion
        m_capTime = 100 * dd4hep::ns;
```